src/styles/MuiThemeProvider.js

```diff
@@ -1,20 +1,29 @@
 import {Component, PropTypes} from 'react';
 import getMuiTheme from './getMuiTheme';
+import getUniqueIdGenerator from '../utils/getUniqueIdGenerator';
+
 
 class MuiThemeProvider extends Component {
 
   static propTypes = {
     children: PropTypes.element,
     muiTheme: PropTypes.object,
+    uniqueIdGen: PropTypes.func,
   };
 
   static childContextTypes = {
     muiTheme: PropTypes.object.isRequired,
+    uniqueIdGen: PropTypes.func.isRequired,
+  };
+
+  state = {
+    uniqueIdGen: this.props.uniqueIdGen || getUniqueIdGenerator(),
```

Comment:
I see, let's not update it in this case. It looks like we have reached the limitation of this solution.
I can't see real use cases for this edge case, I think that we are fine.
Should we add a comment?

*P.S. I start to feel like using an heuristic for the id generation would have been simpler.* 😁